### PR TITLE
Test: deposit test

### PR DESCRIPTION
**Pull request by haythem96**
# Task: deposit test

**Should be merged after #690**

## Description

This PR add a second test case for `deposit()`

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Please describe how to test to verify the changes. Provide instructions so we can reproduce.

## FE Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

## User Facing Checklist

- [ ] I fully understand the user problem this PR is solving
- [ ] I know who the target user is for this PR and have a deep understanding of that user
- [ ] I have tried this flow thinking from the pov of the target user for this PR
- [ ] I (or working w someone on team) have scheduled a user test for this PR (if it is a large change)

### CODE REVIEW

**Comment by haythem96** [comment]
Let's try to make code coverage, branches % 90% or more
Screen Shot 2022-10-30 at 11 44 28 PM

File: packages/bull-vault/test/e2e/BullStrategyTestFork.t.sol [hunk 1]
```diff
@@ -96,6 +96,32 @@ contract BullStrategyTestFork is Test {
...
+    function testSecondDeposit() public {
+        uint256 crabToDepositInitially = 10e18;
+
+        vm.startPrank(user1);
+        (uint256 wethToLend, uint256 usdcToBorrow) = _deposit(crabToDepositInitially);
+        vm.stopPrank();
+
+        assertEq(bullStrategy.balanceOf(user1), crabToDepositInitially);
+        assertEq(IEulerDToken(dToken).balanceOf(address(bullStrategy)), usdcToBorrow);
+        assertTrue(wethToLend.sub(IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy))) <= 1);
```
Comment (daryakaviani) [inline packages/bull-vault/test/e2e/BullStrategyTestFork.t.sol:108]: This is fine for this test, but if you add a fuzzing one, note that sometimes it gets to 2 wei difference.

File: packages/bull-vault/test/e2e/BullStrategyTestFork.t.sol [hunk 2]
```diff
@@ -96,6 +96,32 @@ contract BullStrategyTestFork is Test {
...
+        uint256 userBullBalanceBefore = bullStrategy.balanceOf(user1);
+        uint256 crabToDepositSecond = 7e18;
+        uint256 bullToMint = _calcBullToMint(crabToDepositSecond);
+        vm.startPrank(user1);
+        (uint256 wethToLendSecond, uint256 usdcToBorrowSecond) = _deposit(crabToDepositSecond);
+        vm.stopPrank();
+
+        assertEq(bullStrategy.balanceOf(user1).sub(userBullBalanceBefore), bullToMint);
+        assertEq(IEulerDToken(dToken).balanceOf(address(bullStrategy)).sub(usdcToBorrow), usdcToBorrowSecond);
+        assertTrue(wethToLendSecond.sub(IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy)).sub(wethToLend)) <= 1);
```
Comment (daryakaviani) [inline packages/bull-vault/test/e2e/BullStrategyTestFork.t.sol:121]: This is fine for this test, but if you add a fuzzing one, note that sometimes it gets to 2 wei difference.

**Review by daryakaviani — APPROVED** [review]
lgtm, see note about fuzzing in future

File: packages/bull-vault/test/e2e/BullStrategyTestFork.t.sol (hunk 2 above)
Comment (haythem96) [reply packages/bull-vault/test/e2e/BullStrategyTestFork.t.sol:121]: Added issue https://github.com/opynfinance/squeeth-monorepo/issues/696

**Comment by aleone** [comment]
looks good

**Pull request merged**